### PR TITLE
Silence a lgtm warning

### PR DESCRIPTION
**Pull request by kevingranade**
#### Summary
SUMMARY: None

#### Purpose of change
lgtm.com warnings are a mixture of noise, best practices, and bugs.  The signal to noise ratio seems low enough that eliminating all of them is worthwhile.

#### Describe the solution
These are misc fixes that fall into several categories.
Calling c++-style math functions instead of c-style, i.e. std::round() instead of round().
Renaming a shadowing variable.

#### Testing
Wait for a lgtm.com run to confirm the fixes, most of these should be strict refactors.

### CODE REVIEW

File: src/activity_handlers.cpp [hunk 1]
```diff
@@ -4710,7 +4710,7 @@ void activity_handlers::spellcasting_finish( player_activity *act, player *p )
 {
     act->set_to_null();
     const int level_override = act->get_value( 0 );
-    spell &casting = player_or_item_spell( p, spell_id( act->name ), level_override );
+    spell &spell_being_cast = player_or_item_spell( p, spell_id( act->name ), level_override );
```
Comment (KorGgenT) [inline src/activity_handlers.cpp:4713]: i thought i fixed this one, but then i realized it's still a draft pt

Comment (KorGgenT) [reply src/activity_handlers.cpp:4713]: https://github.com/CleverRaven/Cataclysm-DDA/pull/39101

Comment (kevingranade) [reply src/activity_handlers.cpp:4713]: Aha, yea I can back that out if you have it or you can rebase if that's easier for you, just let me know.

File: src/activity_handlers.cpp [hunk 2]
```diff
@@ -4747,8 +4747,8 @@ void activity_handlers::spellcasting_finish( player_activity *act, player *p )
 {
     act->set_to_null();
     const int level_override = act->get_value( 0 );
-    spell casting;
-    player_or_item_spell( p, spell_id( act->name ), level_override, casting );
+    spell &spell_being_cast;
```
Comment (ZhilkinSerg) [inline src/activity_handlers.cpp:4750]:
```suggestion
    spell spell_being_cast;
```

File: src/activity_handlers.cpp [hunk 3]
```diff
@@ -4757,14 +4757,14 @@ void activity_handlers::spellcasting_finish( player_activity *act, player *p )
     target_handler th;
     tripoint target = p->pos();
     bool target_is_valid = false;
-    if( casting.range() > 0 && !casting.is_valid_target( target_none ) &&
+    if( spell_being_cast.range() > 0 && !spell_being_cast.is_valid_target( target_none ) &&
         !casting.has_flag( RANDOM_TARGET ) ) {
```
Comment (ZhilkinSerg) [inline src/activity_handlers.cpp:4761]:
```suggestion
        !spell_being_cast.has_flag( RANDOM_TARGET ) ) {
```

**Comment by kevingranade** [comment]
This pull request has been mentioned on **Cataclysm: Dark Days Ahead**. There might be relevant details there:

https://discourse.cataclysmdda.org/t/magiclysm-a-total-fantasy-overhaul/20612/323

**Pull request merged**